src/runtime_src/core/pcie/linux/shim.cpp

```diff
@@ -2665,6 +2666,9 @@ int xclP2pEnable(xclDeviceHandle handle, bool enable, bool force)
     xocl::shim *drv = xocl::shim::handleCheck(handle);
     return drv ? drv->p2pEnable(enable, force) : -ENODEV;
   }
+  catch(const xrt_core::error&) {
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
```

Comment:
This is not friendly to the caller.  You are hiding the actual exception.

Reply:
I moved the exception parsing out of the shim.cpp file and partly into the ishim.h file. It is very strange how we throw exceptions and return error values from p2pEnable, is that the expected style? It seems like most of the functions in shim.cpp return error codes, but p2pEnable throws exceptions and returns some codes. Maybe this should be looked at some more? Anyways let me know if the current changes makes a bit more sense.